raiden-ts/src/transport/epics/helpers.ts

```diff
@@ -76,70 +127,29 @@ export function getRoom$(matrix: MatrixClient, roomIdOrAlias: string): Observabl
   allowRtc = false,
 ): Observable<string> {
   const RETRY_COUNT = 3; // is this relevant enough to become a constant/setting?
-  return latest$.pipe(
-    filter(({ presences }) => address in presences),
-    take(1),
-    mergeMap(({ rtc }) => {
-      // if available & open, use channel
-      if (allowRtc && rtc?.[address]?.readyState === 'open') return of(rtc[address]);
-      // else, wait for member to join in the first room, and return roomId
-      return latest$.pipe(
-        map(({ state }) => state.transport.rooms?.[address]?.[0]),
-        // wait for a room to exist (created or invited) for address
-        filter(isntNil),
-        distinctUntilChanged(),
-        // this switchMap unsubscribes from previous "wait" if first room for address changes
-        switchMap((roomId) =>
-          // get/wait room object for roomId
-          // may wait for the room state to be populated (happens after createRoom resolves)
-          getRoom$(matrix, roomId).pipe(
-            mergeMap((room) =>
-              // wait for address to be monitored & online (after getting Room for address)
-              // latest$ ensures it happens immediatelly if all conditions are satisfied
-              latest$.pipe(
-                pluck('presences', address),
-                map((presence) =>
-                  presence?.payload?.available ? presence.payload.userId : undefined,
-                ),
-                distinctUntilChanged(),
-                map((userId) => ({ room, userId })),
-              ),
-            ),
-            // when user is online, get room member for partner's userId
-            // this switchMap unsubscribes from previous wait if userId changes or go offline
-            switchMap(({ room, userId }) => {
-              if (!userId) return EMPTY; // user not monitored or not available
-              const member = room.getMember(userId);
-              // if it already joined room, return its membership
-              if (member && member.membership === 'join') return of(member);
-              // else, wait for the user to join/accept invite
-              return fromEvent<RoomMember>(
-                matrix,
-                'RoomMember.membership',
-                ({}: MatrixEvent, member: RoomMember) => member,
-              ).pipe(
-                filter(
-                  (member) =>
-                    member.roomId === room.roomId &&
-                    member.userId === userId &&
-                    member.membership === 'join',
-                ),
-              );
-            }),
-            pluck('roomId'),
-          ),
-        ),
-      );
-    }),
+  return merge(
+    // if available & open, use channel
+    latest$.pipe(
+      filter(
+        ({ presences, rtc }) =>
+          allowRtc &&
+          address in presences &&
+          presences[address].payload.available &&
+          rtc[address]?.readyState === 'open',
```

Comment:
Here it would be nice to use a constant

Reply:
This is a constant, just a literal in-line one. In our codestyle, we don't favor declaring new variables for every single literal we use throughout the code, only if it's reused/exported or semantically meaningful. I don't think this is the case here.